### PR TITLE
Use OUTPUT_FILES instead of header-postscript.sh

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
This replaces `header-postscript.sh` using the new `OUTPUT_FILES` variable and moves some small headers into the `TEST_OUTPUT` section of their respective test (because they don't need to be separate files anymore).

Best reviewed per commit/test.

~~Note: This currently includes the fix from #10917.~~

### CODE REVIEW

**Comment by Geod24** [comment]
Auto-tester complains because path differ on Windows

**Review by wilzbach — APPROVED** [review]
Though it needs a rebase.

File: test/tools/d_do_test.d [hunk 1]
```diff
@@ -561,6 +561,10 @@ bool gatherTestParameters(ref TestArgs testArgs, string input_dir, string input_
         testArgs.compileOutput = readText(testArgs.compileOutputFile)
                                     .unifyNewLine() // Avoid CRLF issues
                                     .strip();
+
+        // Only sanitize separators from file types that support standalone \
+        if (!testArgs.compileOutputFile.endsWith(".json"))
+            testArgs.compileOutput = testArgs.compileOutput.unifyDirSep(envData.sep);
```
Comment (MoonlightSentinel) [inline test/tools/d_do_test.d:567]: Not sure if DMD should use `\\` in the JSON output but we can always remove this if the output is changed.

**Comment by MoonlightSentinel** [comment]
Resolved the issue w.r.t. mismatched directory separators on Windows

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Hmm, doc tester seems stuck

**Comment by MoonlightSentinel** [comment]
Restarted the tests

**Comment by MoonlightSentinel** [comment]
Doc tester finally passed @thewilsonator

**Pull request merged**